lib/options_validator.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+
+const isObject = require('./utils').isObject;
+
+let validationLevel = 'warn';
+
+function createOptionsSchema(optionsSchema) {
+  return validationFunction.bind(optionsSchema);
+}
+
+function validationFunction(providedOptions) {
+  let validatedOptions = Object.assign({}, providedOptions);
+
+  Object.keys(validatedOptions).forEach(option => {
```

Comment:
nit: `Object.keys(...).forEach` used to be much slower than a for loop over the keys, might be worth evaluating the performance difference